src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -832,9 +832,20 @@ object ICustomAstVisitor.VisitParamBlock(ParamBlockAst paramBlockAst)
         object ICustomAstVisitor.VisitNamedBlock(NamedBlockAst namedBlockAst)
         {
             var inferredTypes = new List<PSTypeName>();
-            for (var index = 0; index < namedBlockAst.Statements.Count; index++)
+            for (int index = 0; index < namedBlockAst.Statements.Count; index++)
             {
-                var ast = namedBlockAst.Statements[index];
+                StatementAst ast = namedBlockAst.Statements[index];
+                if (ast is AssignmentStatementAst
+                    || (ast is PipelineAst pipe && pipe.PipelineElements.Count == 1 && pipe.PipelineElements[0] is CommandExpressionAst cmd
+                    && cmd.Redirections.Count == 0 && cmd.Expression is UnaryExpressionAst unary
+                    && unary.TokenKind is TokenKind.PostfixPlusPlus or TokenKind.PlusPlus or TokenKind.PostfixMinusMinus or TokenKind.MinusMinus))
+                {
+                    // Assignments don't output anything to the named block unless they are wrapped in parentheses.
+                    // When they are wrapped in parentheses, they are seen as PipelineAst.
+                    // Increment/decrement operators like $i++ also don't output anything unless there's a redirection, or they are wrapped in parentheses.
```

Comment:
I tried `++` and I don't see any changes before and after.

Reply:
Try this: `(TabExpansion2 '$i = 1; $Res = if ($true) {$i++}; $Res.').CompletionMatches` in my retail 7.5 build it produces completion results for an int, whereas in my own branch it results in nothing because it correctly infers that there's no output.

Reply:
Added counts helps to understand tests better. Thanks!